sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/AzMonListExtensions.cs

```diff
@@ -239,6 +239,20 @@ internal static string GetDefaultDbPort(string? dbSystem)
             return target;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static string? GetTargetUsingServerAttributes(this AzMonList tagObjects, string defaultPort)
+        {
+            var values = AzMonList.GetTagValues(ref tagObjects, SemanticConventions.AttributeServerAddress, SemanticConventions.AttributeServerSocketAddress, SemanticConventions.AttributeServerPort);
+            string? target = values[0]?.ToString() ?? values[1]?.ToString();
```

Comment:
Should this be followed on Http and Messaging semantics too?

Reply:
Not sure, if I follow. Could you elaborate please?

Reply:
Here we check server.address. If it is null, we take a look at server.socketaddress. Should we follow the same pattern for HTTP and Messaging? In HTTP, we just check for server.address.

Reply:
got it. This is based off of what instrumentation libraries are setting. In case of SQL, there is a possibility that we would set socket.address attribute. In case of http we are not doing that. Need to check messaging spec to see if it makes sense to add it there.